pkg/api/pod/util.go

```diff
@@ -795,6 +798,28 @@ func podLifecycleSleepActionInUse(podSpec *api.PodSpec) bool {
 	return inUse
 }
 
+func podLifecycleSleepActionZeroValueInUse(podSpec *api.PodSpec) bool {
+	if podSpec == nil {
+		return false
+	}
+	var inUse bool
+	VisitContainers(podSpec, AllContainers, func(c *api.Container, containerType ContainerType) bool {
```

Comment:
Lifecycle is not allowed for ephemeral containers.
- However, `c.Lifecycle == nil` will return true for ephemeral containers.

Reply:
Yes, for ephemeral containers `podLifecycleSleepActionZeroValueInUse` will return false after exiting from `VisitContainers`.